From: Danny Kukawka <danny.kukawka@bisect.de>
Date: Tue, 21 Feb 2012 02:07:49 +0000
Subject: [PATCH 071/105] ethernet: unify return value of .ndo_set_mac_address
 if address is invalid
Patch-mainline: v3.4-rc1
Git-commit: 504f9b5a6bb5336ad434438d0cdd61a16db80129 (partial)
References: bnc#790588 FATE#313912

Unify return value of .ndo_set_mac_address if the given address
isn't valid. Return -EADDRNOTAVAIL as eth_mac_addr() already does
if is_valid_ether_addr() fails.

Signed-off-by: Danny Kukawka <danny.kukawka@bisect.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Nithin Nayak Sujir <nsujir@broadcom.com>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/net/ethernet/broadcom/tg3.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -8129,7 +8129,7 @@ static int tg3_set_mac_addr(struct net_d
 	int err = 0, skip_mac_1 = 0;
 
 	if (!is_valid_ether_addr(addr->sa_data))
-		return -EINVAL;
+		return -EADDRNOTAVAIL;
 
 	memcpy(dev->dev_addr, addr->sa_data, dev->addr_len);
 
